File: Task2/task2.ts
import * as fs from 'fs';
const input = require('fs').readFileSync(0, 'utf-8').trim().split('\n');
let numbers = +input[0];
if (numbers <= 0) {
    console.log("Быть не может");
    process.exit(0);
}
let clumba: number[] = [];
let max = 0;
for (let i = 1; i <= numbers; ++i) {
    let flower = +input[i];
    if (flower > 0) {
        clumba.push(flower);
    }
    if (flower > max) {
        max = flower;
    }
}
let poliv = +input[numbers + 1];
if (poliv < max) {
    console.log("Как купишь новую лейку - приходи.");
    process.exit(0);
}
let step = 0;
let perem = poliv;
for (let i = 0; i < clumba.length; i++) {
    if (perem < clumba[i]) {
        step += (i * 2 + 1);
        perem = poliv;
    } else {
        step += 1;
    }
    perem -= clumba[i];
}
console.log(step);

File: Task2/task2.rs
use std::io;
fn main() {
    let mut numbers = String::new();
    io::stdin().read_line(&mut numbers).unwrap();
    let number: i32 = match numbers.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Неправильный ввод.");
            return;
        }
    };
    if number <= 0 {
        println!("Быть не может");
        return;
    }
    let mut max = 0;
    let mut clumba = Vec::new();
    for _i in 0..number{
        let mut flow = String::new();
        io::stdin().read_line(&mut flow).unwrap();
        let flower: i32 = match flow.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                println!("Неправильный ввод.");
                continue;
            }
        };
        if flower > 0{
            clumba.push(flower); 
        }
        if flower > max{
            max = flower;
        }
    }
    let mut leika = String::new();
    io::stdin().read_line(&mut leika).unwrap();
    let poliv: i32 = match leika.trim().parse() {
        Ok(num) => num,
        Err(_) => {
            println!("Неправильный ввод.");
            return;
        }
    };
    if poliv < max {
        println!("Как купишь новую лейку - приходи.");
        return;
    }
    let mut step = 0;
    let mut perem = poliv;
    for i in 0..clumba.len(){
        if perem < clumba[i]
        {
            step = step + (i * 2 + 1);
            perem = poliv;
        }
        else
        {
            step += 1;
        }
        perem -= clumba[i];
    }
    println!("{}", step);
}
